src/sidebar/test/groups-test.js

```diff
@@ -58,121 +68,156 @@ describe('groups', function() {
   });
 
   function service() {
-    return groups(fakeLocalStorage, fakeServiceUrl, fakeSession,
+    return groups(fakeAnnotationUI, fakeLocalStorage, fakeServiceUrl, fakeSession,
       fakeRootScope, fakeStore);
   }
 
   describe('.all()', function() {
     it('returns no groups if there are none in the session', function() {
-      fakeSession = {state: {groups: []}};
+      fakeSession = {state: {}};
 
       var groups = service().all();
 
       assert.equal(groups.length, 0);
     });
 
-    it('returns the groups from the session when there are some', function() {
-      var groups = service().all();
+    it('returns the groups when there are some', function() {
+      var svc = service();
+
+      return svc.load().then(() => {
+        var groups = svc.all();
+        assert.equal(groups.length, 3);
+        assert.deepEqual(groups, [
+          {name: 'Group 1', id: 'id1'},
+          {name: 'Group 2', id: 'id2'},
+          {name: 'Group 3', id: 'id3'},
+        ]);
+      });
+    });
+  });
+
+  describe('.load() method', function() {
```

Comment:
Some of our tests are not consistent about this, but the canonical naming for `describe` blocks for methods is `describe('#someMethod')`, taken from JSDoc conventions for referencing instance methods/properties.